### ISSUE TITLE
Runtime Panic - amazon-ebs: Error deleting volume

### ISSUE THREAD

**Issue by alexmac** [mitchellh/packer#4401]
- Packer version/Host:

```
2017/01/14 08:12:05 [INFO] Packer version: 0.12.1
2017/01/14 08:12:05 Packer Target OS/Arch: linux amd64
2017/01/14 08:12:05 Built with Go Version: go1.7.3
```
crash log output:

```
2017/01/14 08:16:18 ui: ==> amazon-ebs: Terminating the source AWS instance...
2017/01/14 08:16:18 packer: 2017/01/14 08:16:18 Waiting for state to become: terminated
2017/01/14 08:16:18 packer: 2017/01/14 08:16:18 Using 2s as polling delay (change with AWS_POLL_DELAY_SECONDS)
2017/01/14 08:16:18 packer: 2017/01/14 08:16:18 Allowing 300s to complete (change with AWS_TIMEOUT_SECONDS)
2017/01/14 08:16:24 ui: ==> amazon-ebs: Cleaning up any extra volumes...
2017/01/14 08:16:25 ui: ==> amazon-ebs: Destroying volume (vol-0cb709c5e6a7a085d)...
2017/01/14 08:16:25 ui: ==> amazon-ebs: Error deleting volume: vol-0cb709c5e6a7a085d
2017/01/14 08:16:25 packer: panic: runtime error: invalid memory address or nil pointer dereference
2017/01/14 08:16:25 packer: [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x5ecc57]
2017/01/14 08:16:25 packer: 
2017/01/14 08:16:25 packer: goroutine 26 [running]:
2017/01/14 08:16:25 packer: panic(0x13beea0, 0xc420014080)
2017/01/14 08:16:25 packer: 	/usr/local/Cellar/go/1.7.3/libexec/src/runtime/panic.go:500 +0x1a1
2017/01/14 08:16:25 packer: github.com/mitchellh/packer/builder/amazon/ebs.(*stepCreateAMI).Run(0xc42016a2d0, 0x1e117a0, 0xc4203b8600, 0x0)
2017/01/14 08:16:25 packer: 	/Users/mwhooker/go/src/github.com/mitchellh/packer/builder/amazon/ebs/step_create_ami.go:72 +0x917
2017/01/14 08:16:25 packer: github.com/mitchellh/packer/vendor/github.com/mitchellh/multistep.(*BasicRunner).Run(0xc420371440, 0x1e117a0, 0xc4203b8600)
2017/01/14 08:16:25 packer: 	/Users/mwhooker/go/src/github.com/mitchellh/packer/vendor/github.com/mitchellh/multistep/basic_runner.go:70 +0x207
2017/01/14 08:16:25 packer: github.com/mitchellh/packer/builder/amazon/ebs.(*Builder).Run(0xc42027cf00, 0x1e140e0, 0xc4203b5840, 0x1e0d0a0, 0xc4203a59f0, 0x1e12e60, 0xc42016a290, 0xc420089dd0, 0x0, 0x0, ...)
2017/01/14 08:16:25 packer: 	/Users/mwhooker/go/src/github.com/mitchellh/packer/builder/amazon/ebs/builder.go:196 +0x11cb
2017/01/14 08:16:25 packer: github.com/mitchellh/packer/packer/rpc.(*BuilderServer).Run(0xc4200e36e0, 0x1, 0xc42008dbe0, 0x0, 0x0)
2017/01/14 08:16:25 packer: 	/Users/mwhooker/go/src/github.com/mitchellh/packer/packer/rpc/builder.go:93 +0x218
2017/01/14 08:16:25 packer: reflect.Value.call(0xc42008e7e0, 0xc420082268, 0x13, 0x15ffd31, 0x4, 0xc420551ed0, 0x3, 0x3, 0x0, 0x0, ...)
2017/01/14 08:16:25 packer: 	/usr/local/Cellar/go/1.7.3/libexec/src/reflect/value.go:434 +0x5c8
2017/01/14 08:16:25 packer: reflect.Value.Call(0xc42008e7e0, 0xc420082268, 0x13, 0xc420551ed0, 0x3, 0x3, 0xc420325138, 0x1, 0x0)
2017/01/14 08:16:25 packer: 	/usr/local/Cellar/go/1.7.3/libexec/src/reflect/value.go:302 +0xa4
2017/01/14 08:16:25 packer: net/rpc.(*service).call(0xc42008bd80, 0xc42008bd40, 0xc4202ec090, 0xc4200ac880, 0xc42029d560, 0x13117c0, 0xc42008dafc, 0x18a, 0x12e6d00, 0xc42008dbe0, ...)
2017/01/14 08:16:25 packer: 	/usr/local/Cellar/go/1.7.3/libexec/src/net/rpc/server.go:383 +0x148
2017/01/14 08:16:25 packer: created by net/rpc.(*Server).ServeCodec
2017/01/14 08:16:25 packer: 	/usr/local/Cellar/go/1.7.3/libexec/src/net/rpc/server.go:477 +0x421
2017/01/14 08:16:25 ui error: Build 'amazon-ebs' errored: unexpected EOF
2017/01/14 08:16:25 Builds completed. Waiting on interrupt barrier...
2017/01/14 08:16:25 machine readable: error-count []string{"1"}
2017/01/14 08:16:25 ui error: 
==> Some builds didn't complete successfully and had errors:
2017/01/14 08:16:25 machine readable: amazon-ebs,error []string{"unexpected EOF"}
2017/01/14 08:16:25 ui error: --> amazon-ebs: unexpected EOF
2017/01/14 08:16:25 ui: 
==> Builds finished but no artifacts were created.
2017/01/14 08:16:25 waiting for all plugin processes to complete...
2017/01/14 08:16:25 /usr/local/bin/packer: plugin process exited
2017/01/14 08:16:25 /usr/local/bin/packer: plugin process exited
2017/01/14 08:16:25 /usr/local/bin/packer: plugin process exited
2017/01/14 08:16:25 /usr/local/bin/packer: plugin process exited
2017/01/14 08:16:25 /usr/local/bin/packer: plugin process exited
```

**Comment by rickard-von-essen** [comment]
Please supply the information requested in the issue template:
- Packer version (`packer version`)
- Host platform (`uname -a` etc.)
- Debug log output from `PACKER_LOG=1 packer build template.json`.
  Please paste this in a [gist](https://gist.github.com).
- The _simplest example template and scripts_ needed to reproduce the bug.
  Include these in your [gist](https://gist.github.com).

**Comment by alexmac** [comment]
The above was from our linux build machine, but I can repro it locally on my mac:

Packer version: 0.12.1
Host: Darwin 16.3.0 Darwin Kernel Version 16.3.0: Thu Nov 17 20:23:58 PST 2016; root:xnu-3789.31.2~1/RELEASE_X86_64 x86_64

debug output: https://gist.github.com/alexmac/4a2b5a79df23578546567b1736d76bc9
repro json: https://gist.github.com/alexmac/e325e9a45782e643a6f4bb3cae48e56e

**Comment by rickard-von-essen** [comment]
Thanks for reporting this. This is a duplicate of #4303, which have been fixed on master.
